### PR TITLE
Include arm64 libtorch in download instructions with auto-generated page

### PR DESCRIPTION
**Pull request by atalman**
MacOS x86 and Macos Arm64 instructions used to be exactly the same, hence we generated all the instructions from MacOS x86. Now since we added arm64 builds for libtorch in nightly these instructions need to get into the getting started page.

### CODE REVIEW

File: scripts/gen_quick_start_module.py [hunk 1]
```diff
@@ -58,8 +59,8 @@ class OperatingSystem(Enum):
         CXX11_ABI: "Download here (cxx11 ABI):",
         RELEASE: "Download here (Release version):",
         DEBUG: "Download here (Debug version):",
-        MACOS: "Download x86 libtorch here (ROCm and CUDA are not supported):",
-        MACOS_ARM64: "Download arm64 libtorch here (ROCm and CUDA are not supported):",
+        MACOS: "Download x86 libtorch here",
+        MACOS_ARM64: "Download arm64 libtorch here",
```
Comment (malfet) [inline scripts/gen_quick_start_module.py:63]: Can you please submit this change as a separate PR (and I want Greg to have a quick look at it)

File: scripts/gen_quick_start_module.py [hunk 2]
```diff
@@ -123,6 +124,12 @@ def update_versions(versions, release_matrix, release_version):
                     if (x["package_type"], x["gpu_arch_type"], x["gpu_arch_version"]) ==
                     (package_type, gpu_arch_type, gpu_arch_version)
                     ]
+                if os_key == OperatingSystem.MACOS.value and package_type == "libtorch":
+                    pkg_arch_matrix += [
+                        x for x in release_matrix[OperatingSystem.MACOS_ARM64.value]
+                        if (x["package_type"], x["gpu_arch_type"], x["gpu_arch_version"]) ==
+                        (package_type, gpu_arch_type, gpu_arch_version)
+                        ]
```
Comment (malfet) [inline scripts/gen_quick_start_module.py:132]: Can you please add some sort of a comment to explain why those complications are necessary?

File: scripts/gen_quick_start_module.py [hunk 3]
```diff
@@ -142,12 +149,11 @@ def update_versions(versions, release_matrix, release_version):
...
-                            rel_entry_dict = {
-                                x["devtoolset"]: x["installation"] for x in pkg_arch_matrix
-                                if x["libtorch_variant"] == "shared-with-deps"
-                                }
                             if instr["versions"] is not None:
-                                instr["versions"][LIBTORCH_DWNL_INSTR[MACOS]] = list(rel_entry_dict.values())[0]
+                                instr["versions"][LIBTORCH_DWNL_INSTR[MACOS]] = pkg_arch_matrix[0]["installation"]
+                                # todo: remove this once we release libtorch for arm64 as stable
+                                if release_version == "nightly":
+                                    instr["versions"][LIBTORCH_DWNL_INSTR[MACOS_ARM64]] = pkg_arch_matrix[1]["installation"]
```
Comment (malfet) [inline scripts/gen_quick_start_module.py:156]: This feel wrong, it should add those instructions of more flavors are available for the platform, rather than hardcode  MacOS or anything like that here

**Review by malfet — CHANGES_REQUESTED** [review]

File: scripts/gen_quick_start_module.py (hunk 2 above)
Comment (atalman) [reply scripts/gen_quick_start_module.py:132]: Ok Will do. Basically we are trying to reduce 2 Matrices MACOS and MACOS_ARM64 to 1 MacOS matrix that we display to user. This is only applicable to MacOS and currently only to libtorch.

File: scripts/gen_quick_start_module.py (hunk 1 above)
Comment (atalman) [reply scripts/gen_quick_start_module.py]: Added ROCM and CUDA not supported back

File: scripts/gen_quick_start_module.py (hunk 3 above)
Comment (atalman) [reply scripts/gen_quick_start_module.py]: Removed nightly check

**Review by malfet — APPROVED** [review]
This is somewhat ugly, but sure, if it works then it it works...
